Question 1: Download and load the Ames CSV file. Read the file into a Pandas DataFrame and display the first five rows.  

In [2]:
import pandas as pd
df =pd.read_csv(r'C:\Users\bsila\Dexian_Bootcamp\code\DataPyLabExercise\DA_labs\AmesHousing.csv\AmesHousing.csv')
df.head()

Order        PID  MS SubClass MS Zoning  Lot Frontage  Lot Area Street  \
0      1  526301100           20        RL         141.0     31770   Pave   
1      2  526350040           20        RH          80.0     11622   Pave   
2      3  526351010           20        RL          81.0     14267   Pave   
3      4  526353030           20        RL          93.0     11160   Pave   
4      5  527105010           60        RL          74.0     13830   Pave   

  Alley Lot Shape Land Contour  ... Pool Area Pool QC  Fence Misc Feature  \
0   NaN       IR1          Lvl  ...         0     NaN    NaN          NaN   
1   NaN       Reg          Lvl  ...         0     NaN  MnPrv          NaN   
2   NaN       IR1          Lvl  ...         0     NaN    NaN         Gar2   
3   NaN       Reg          Lvl  ...         0     NaN    NaN          NaN   
4   NaN       IR1          Lvl  ...         0     NaN  MnPrv          NaN   

  Misc Val Mo Sold Yr Sold Sale Type  Sale Condition  SalePrice  
0        0       5    2010       WD           Normal     215000  
1        0       6    2010       WD           Normal     105000  
2    12500       6    2010       WD           Normal     172000  
3        0       4    2010       WD           Normal     244000  
4        0       3    2010       WD           Normal     189900  

[5 rows x 82 columns]

Question 2:  Print the highest and lowest priced homes in Ames.

In [3]:
highest_price = df['SalePrice'].max()
print(f"Highest price: ${highest_price}")
Lowest_price= df['SalePrice'].min()
print(f"Lowest price: ${Lowest_price}")

Highest price: $755000
Lowest price: $12789


Question 3:  How many categories does lot size have? 

In [4]:
df.groupby(['Lot Shape']).count()

Order   PID  MS SubClass  MS Zoning  Lot Frontage  Lot Area  \
Lot Shape                                                                
IR1          979   979          979        979           654       979   
IR2           76    76           76         76            48        76   
IR3           16    16           16         16            11        16   
Reg         1859  1859         1859       1859          1727      1859   

           Street  Alley  Land Contour  Utilities  ...  Pool Area  Pool QC  \
Lot Shape                                          ...                       
IR1           979     27           979        979  ...        979        5   
IR2            76      4            76         76  ...         76        0   
IR3            16      1            16         16  ...         16        1   
Reg          1859    166          1859       1859  ...       1859        7   

           Fence  Misc Feature  Misc Val  Mo Sold  Yr Sold  Sale Type  \
Lot Shape                                                               
IR1          163            37       979      979      979        979   
IR2            7             4        76       76       76         76   
IR3            1             0        16       16       16         16   
Reg          401            65      1859     1859     1859       1859   

           Sale Condition  SalePrice  
Lot Shape                             
IR1                   979        979  
IR2                    76         76  
IR3                    16         16  
Reg                  1859       1859  

[4 rows x 81 columns]

Question 4:  What is the range of years for the house listings in the data? 

In [5]:
df[['Year Built', 'Yr Sold', 'Year Remod/Add']].max()

Year Built        2010
Yr Sold           2010
Year Remod/Add    2010
dtype: int64

In [6]:
df[['Year Built', 'Yr Sold', 'Year Remod/Add']].min()

Year Built        1872
Yr Sold           2006
Year Remod/Add    1950
dtype: int64

The range of years for the house listings data is between 1872 and 2010.

Question 5:  What is the total number of all homes sold in Ames for all years? 

In [7]:
df['Yr Sold'].count()

2930

Question 6:  How many nulls or NA are there in the Lot Frontage column? 

In [8]:
df['Lot Frontage'].isna().sum()

490

There are 490 nulls or NA in the Lot Frontage Column

Question 7:  How many items have a value in Alley and what is the total number of  NA in the column? 

In [11]:
df['Alley'].count()

198

In [12]:
df['Alley'].isnull().sum()

2732

There are 198 items that has values in Alley and there are 2732 NA values in the column.

Question 8:  What are all the homes in Gilbert with Yr Sold being 2009 and 2010?

In [14]:
df[(df['Neighborhood'] == 'Gilbert') & df['Yr Sold'].isin([2009, 2010])]

Order        PID  MS SubClass MS Zoning  Lot Frontage  Lot Area Street  \
4        5  527105010           60        RL          74.0     13830   Pave   
5        6  527105030           60        RL          78.0      9978   Pave   
9       10  527162130           60        RL          60.0      7500   Pave   
10      11  527163010           60        RL          75.0     10000   Pave   
11      12  527165230           20        RL           NaN      7980   Pave   
12      13  527166040           60        RL          63.0      8402   Pave   
13      14  527180040           20        RL          85.0     10176   Pave   
16      17  527225035           50        RL         152.0     12134   Pave   
18      19  527276150           20        RL         140.0     19138   Pave   
51      52  528218150           20        RL         100.0     18494   Pave   
54      55  528231020           80        RL          67.0     13300   Pave   
55      56  528240070           60        RL           NaN      7851   Pave   
56      57  528240130           60        RL          63.0      8577   Pave   
57      58  528250100           80        RL           NaN      7750   Pave   
58      59  528292020           60        RL           NaN      9505   Pave   
344    345  527105070           60        RL          88.0     10179   Pave   
345    346  527105130           60        RL           NaN     11792   Pave   
346    347  527107130           80        RL          60.0      8400   Pave   
347    348  527108070           20        RL          42.0     14892   Pave   
348    349  527110020           80        RL           NaN      8530   Pave   
353    354  527161040           80        RL          57.0      8923   Pave   
354    355  527162110           60        RL          74.0     10141   Pave   
355    356  527162120           60        RL          60.0      7500   Pave   
356    357  527163040           60        RL          59.0      7837   Pave   
357    358  527163070           60        RL           NaN      9765   Pave   
358    359  527163130           60        RL           NaN      8803   Pave   
359    360  527164020           20        RL          58.0      7250   Pave   
360    361  527164060           60        RL           NaN      9636   Pave   
361    362  527164090           60        RL          65.0      8125   Pave   
362    363  527165130           20        RL           NaN      9248   Pave   
363    364  527166010           60        RL           NaN     10762   Pave   
364    365  527180100           20        RL          99.0     11851   Pave   
369    370  527276160           60        RL          85.0     11900   Pave   
464    465  528218090           60        RL          63.0     10928   Pave   
465    466  528220040           60        RL          57.0      8918   Pave   
466    467  528221060           60        RL         149.0     12589   Pave   
467    468  528221070           60        RL         122.0     11911   Pave   
474    475  528235010           60        RL          71.0      7795   Pave   
475    476  528235090           60        RL           NaN      8068   Pave   
476    477  528235200           80        RL          59.0      9434   Pave   
477    478  528240030           60        RL          62.0      7984   Pave   
478    479  528240060           80        RL           NaN      7750   Pave   
479    480  528240150           60        RL          61.0     10125   Pave   
480    481  528250020           60        RL           NaN      8965   Pave   
481    482  528250040           60        RL           NaN      8174   Pave   
483    484  528275070           60        RL           NaN      8795   Pave   
484    485  528275160           60        RL           NaN     12891   Pave   
485    486  528280230           60        RL           NaN     12224   Pave   
486    487  528290030           80        RL          61.0      9734   Pave   
487    488  528290120           60        RL          6

Question 9:  What is the count of homes sold in Gilbert with Yr Sold being 2010? 

In [18]:
# Count of homes in Gilbert with Yr Sold being  2010
df[(df['Neighborhood']=='Gilbert') & (df['Yr Sold']==2010)].count()


Order             15
PID               15
MS SubClass       15
MS Zoning         15
Lot Frontage      11
                  ..
Mo Sold           15
Yr Sold           15
Sale Type         15
Sale Condition    15
SalePrice         15
Length: 82, dtype: int64

Question 10:  Return the value of the third row, and fifth column.

In [21]:
df.loc[2, 'Lot Frontage']  #3rd row and fifth column

81.0

Question 11:  Remake a new column from the SalePrice that includes data formatted with the $, and commas for the thousands/millions digits. 

In [22]:
df['SalePrice_modified'] = df['SalePrice'].apply(lambda x: '${:,}'.format(x))
df

Order        PID  MS SubClass MS Zoning  Lot Frontage  Lot Area Street  \
0         1  526301100           20        RL         141.0     31770   Pave   
1         2  526350040           20        RH          80.0     11622   Pave   
2         3  526351010           20        RL          81.0     14267   Pave   
3         4  526353030           20        RL          93.0     11160   Pave   
4         5  527105010           60        RL          74.0     13830   Pave   
...     ...        ...          ...       ...           ...       ...    ...   
2925   2926  923275080           80        RL          37.0      7937   Pave   
2926   2927  923276100           20        RL           NaN      8885   Pave   
2927   2928  923400125           85        RL          62.0     10441   Pave   
2928   2929  924100070           20        RL          77.0     10010   Pave   
2929   2930  924151050           60        RL          74.0      9627   Pave   

     Alley Lot Shape Land Contour  ... Pool QC  Fence Misc Feature Misc Val  \
0      NaN       IR1          Lvl  ...     NaN    NaN          NaN        0   
1      NaN       Reg          Lvl  ...     NaN  MnPrv          NaN        0   
2      NaN       IR1          Lvl  ...     NaN    NaN         Gar2    12500   
3      NaN       Reg          Lvl  ...     NaN    NaN          NaN        0   
4      NaN       IR1          Lvl  ...     NaN  MnPrv          NaN        0   
...    ...       ...          ...  ...     ...    ...          ...      ...   
2925   NaN       IR1          Lvl  ...     NaN  GdPrv          NaN        0   
2926   NaN       IR1          Low  ...     NaN  MnPrv          NaN        0   
2927   NaN       Reg          Lvl  ...     NaN  MnPrv         Shed      700   
2928   NaN       Reg          Lvl  ...     NaN    NaN          NaN        0   
2929   NaN       Reg          Lvl  ...     NaN    NaN          NaN        0   

     Mo Sold Yr Sold Sale Type Sale Condition  SalePrice  SalePrice_modified  
0          5    2010       WD          Normal     215000            $215,000  
1          6    2010       WD          Normal     105000            $105,000  
2          6    2010       WD          Normal     172000            $172,000  
3          4    2010       WD          Normal     244000            $244,000  
4          3    2010       WD          Normal     189900            $189,900  
...      ...     ...       ...            ...        ...                 ...  
2925       3    2006       WD          Normal     142500            $142,500  
2926       6    2006       WD          Normal     131000            $131,000  
2927       7    2006       WD          Normal     132000            $132,000  
2928       4    2006       WD          Normal     170000            $170,000  
2929      11    2006       WD          Normal     188000            $188,000  

[2930 rows x 83 columns]

Question 12:   Output a dataframe with only SalePrice, Neighborhood, and Year Built.  Save this file in your git repository as “minimal_ames_housing.csv”,  Make sure you include the formatted SalePrice from the previous question.

In [23]:
df2 = df[['Neighborhood', 'Year Built', 'SalePrice_modified']]
df2

Neighborhood  Year Built SalePrice_modified
0           NAmes        1960           $215,000
1           NAmes        1961           $105,000
2           NAmes        1958           $172,000
3           NAmes        1968           $244,000
4         Gilbert        1997           $189,900
...           ...         ...                ...
2925      Mitchel        1984           $142,500
2926      Mitchel        1983           $131,000
2927      Mitchel        1992           $132,000
2928      Mitchel        1974           $170,000
2929      Mitchel        1993           $188,000

[2930 rows x 3 columns]

In [24]:
df2.to_csv("minimal_ames_housing.csv", index=False)

In [25]:
import os  
os.makedirs(r'C:\Users\bsila\Dexian_Bootcamp\code\DataPyLabExercise\DA_labs\AmesHousing.csv', exist_ok=True)  
df2.to_csv(r'C:\Users\bsila\Dexian_Bootcamp\code\DataPyLabExercise\DA_labs\AmesHousing.csv\minimal_ames_housing.csv')